# LPM-модель: подгонка

In [1]:
import numpy as np
import pandas as pd
import statsmodels.formula.api as smf

In [4]:
# импорт датасета
df = pd.read_csv('loanapp.csv')
df

occ  loanamt  action   msa  suffolk  appinc  typur  unit  married  dep  \
0       1       89       1  1120        0      72      0   1.0      0.0  0.0   
1       1      128       3  1120        0      74      0   1.0      1.0  1.0   
2       1      128       1  1120        0      84      3   1.0      0.0  0.0   
3       1       66       1  1120        0      36      0   1.0      1.0  0.0   
4       1      120       1  1120        0      59      8   1.0      1.0  0.0   
...   ...      ...     ...   ...      ...     ...    ...   ...      ...  ...   
1984    1      158       1  1120        0      96      0   1.0      1.0  0.0   
1985    1       35       1  1120        0     169      1   1.0      1.0  0.0   
1986    2      225       1  1120        0      49      0   2.0      1.0  0.0   
1987    1       98       1  1120        1     110      1   1.0      0.0  0.0   
1988    1      133       3  1120        0      55      0   1.0      0.0  0.0   

      ...  approve  mortno  mortperf  mortlat1  mortlat2  chist  multi  \
0     ...        1       0         1         0         0      1    0.0   
1     ...        0       0         1         0         0      1    0.0   
2     ...        1       0         1         0         0      1    0.0   
3     ...        1       0         1         0         0      0    0.0   
4     ...        1       0         1         0         0      1    0.0   
...   ...      ...     ...       ...       ...       ...    ...    ...   
1984  ...        1       0         1         0         0      1    0.0   
1985  ...        1       1         0         0         0      1    0.0   
1986  ...        1       0         1         0         0      1    1.0   
1987  ...        1       1         0         0         0      1    0.0   
1988  ...        0       0         1         0         0      1    0.0   

       loanprc  thick  white  
0     0.754237    0.0      1  
1     0.800000    1.0      1  
2     0.895105    1.0      1  
3     0.600000    0.0      1  
4     0.895522    0.0      1  
...        ...    ...    ...  
1984  0.897727    0.0      1  
1985  0.111111    0.0      1  
1986  1.000000    0.0      1  
1987  0.455814    0.0      1  
1988  0.898649    0.0      0  

[1989 rows x 59 columns]

## Модель
Для датасета `loanapp`
рассморим регрессию **approve на mortno, unem, dep, male, married, yjob, self**

Спецификация: 
$$
  approve=\beta_0+\beta_1mortno+\beta_2unem+\beta_3dep+\beta_4male+\beta_5married+\beta_6yjob+\beta_7self+u
$$

Альтернативная спецификация:
$$ 
  P(approve=1)=\beta_0+\beta_1mortno+\beta_2unem+\beta_3dep+\beta_4male+\beta_5married+\beta_6yjob+\beta_7self
$$

In [6]:
mod = smf.ols(formula='approve~mortno+unem+dep+male+married+yjob+self', data=df)
res = mod.fit(cov_type='HC3') # подгонка модели с поправкой на гетероскедастиность

In [7]:
# коэффициенты подогнанной модели
res.params

Intercept    0.864212
mortno       0.073251
unem        -0.006434
dep         -0.018472
male         0.001907
married      0.045946
yjob        -0.000662
self        -0.036120
dtype: float64

In [9]:
# коэффициенты с округлением
res.params.round(3)

Intercept    0.864
mortno       0.073
unem        -0.006
dep         -0.018
male         0.002
married      0.046
yjob        -0.001
self        -0.036
dtype: float64

## Интерпретация
- отсуствие ипотечной кредитной истории увеличивает вероятгность "положительного исхода" на 0.073
- увеличение базработцы в отрасли уменьшает вероятгность "положительного исхода" на 0.006
- доплнительный иждевенец уменьшает вероятгность "положительного исхода" на 0.018
- дополнительный год стажа на текущем месте уменьшает вероятгность "положительного исхода" на 0.036